### DISCUSSION TITLE
Kustomize Controller and Private Remote Bases/Resources with HTTPS git url

### DISCUSSION THREAD

**Discussion by beefpine** [fluxcd/flux2#3333] [Q&A]
Apologies if this is a duplicate question/discussion, however despite my searching I haven't found anything to address my particular situation.

I have a private repo owned by another company (repo A), which I am already able to deploy via flux and helm. However, I am looking into integrating kustomize to support the management of unique configurations across different environments (i.e. overlays).

As I cannot modify repo A, I have created my own repo (repo B) to contain the overlays and have its base folder resource section point to the git url for repo A (referencing [remote base documentation](https://github.com/kubernetes-sigs/kustomize/blob/a7f4db7fb41e17b2c826a524f545e6174b4dc6ac/examples/remoteBuild.md)). 
Out of the box this fails in the kustomization controller with an error mentioning failing to accumulate resources and git exit code 128. My search across discussions and posts tells me it is because the repo A is private and the kustomization controller does not have the credentials to authenticate.

I have seen a post dating back to 2020 indicating that kustomize remote repos does not support private repos out of the box (https://github.com/fluxcd/flux2/discussions/326#discussioncomment-96578).

I have also seen posts indicating that patching the kustomize controller with key-based ssh auth works to enable the kustomize controller to authenticate (https://github.com/fluxcd/kustomize-controller/issues/158).

Using the ssh approach, I have been able to have my overlays reside in repo B and applied to repo A via flux successfully.

**However**, I am in a situation where not all environments will have ssh keys available to mount to the kustomize controller. As such, is there a way to have the kustomize controller resolve a https git url via a token?

I have tried referencing https://github.com/fluxcd/flux2/issues/3127, where I define the gitRepository for repo A outside of repo A but still didn't work. I have also tried referencing https://github.com/fluxcd/flux2/discussions/2578 by modifying repo B's gitRepository to have an include section pointing to repo A's gitRepository to no avail.

If anyone has an additional insight, it would be greatly appreciated.

Thanks for your time.

**Comment by dinu-mathai** [comment]
__Problem__ : As of now Kustomization controller(the one handled Kustomization CRD objects) of Flux2 does not have option to refer private git repository.

__Solution/Hack__: We can change the Kustomization controller deployment through helm values.yaml. Run the Kustomization controller process as root user and inject a file /root/.gitconfig - that will replace the private repo URL with basic auth credentials.

Refer: https://github.com/dinumathai/flux-inner-private-repo
